Sources/Logging/TestingLogHandler.swift

```diff
@@ -0,0 +1,298 @@
+import Foundation
+
+#if os(Linux)
+    import Glibc
+#else
+    import Darwin.C
+#endif
+
+public struct LogMessage: CustomStringConvertible {
+    private static let timestampFormat = createTimestampFormat()
+
+    private static func createTimestampFormat() -> DateFormatter {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
+        return dateFormatter
+    }
+
+    public let timestamp: Date
+    public let label: String
+
+    public let level: Logger.Level
+    public let message: Logger.Message
+    public let metadata: Logger.Metadata
+    public let source: String
+
+    public let file: String
+    public let function: String
+    public let line: UInt
+
+    public var prettyMetadata: String {
+        metadata.sorted(by: {
+            $0.key < $1.key
+        }).map {
+            "\($0)=\($1)"
+        }.joined(separator: " ")
+    }
+
+    public var description: String {
+        "\(Self.timestampFormat.string(from: timestamp)) \(level) \(label) :\(prettyMetadata) [\(source)] \(message)"
+    }
+
+    public init(
+        timestamp: Date = Date(),
+        label: String = "nil",
+        level: Logger.Level = .info,
+        message: Logger.Message = .init(stringLiteral: ""),
+        metadata: Logger.Metadata = [:],
+        source: String = "nil",
+        file: String = "nil",
+        function: String = "nil",
+        line: UInt = 0
+    ) {
+        self.timestamp = timestamp
+        self.label = label
+        self.level = level
+        self.message = message
+        self.metadata = metadata
+        self.source = source
+        self.file = file
+        self.function = function
+        self.line = line
+    }
+
+    // Easily check if log message matches certain characteristics:
+    @available(macOS 13.0, *)
+    public func match(
```

Comment:
I am not sure if we need this method at all. Why can't users just write

`#expected(capturedLogMessage.message == "foo")`

Reply:
Typically you would be working with multiple log messages rather than a single captured one.

Perhaps the **real** desire here is to provide a better error message in the event that no single log message matches? For example, if you setup a container so that it only has log entries that match `message == "foo"`, then if no log messages matches, printing an error to the effect of `no log entry matches message == "foo"` would be desirable. Let me see if I can come up with a design for that.

Reply:
I've updated the design proposal so that a new "assertion" method (under the covers it would really just wrap a log "container"). Let me know if you like that idea and I can add that wrapper.